src/core/annotation.js

```diff
@@ -1048,13 +1048,14 @@ class PopupAnnotation extends Annotation {
 
     let parentSubtype = parentItem.get('Subtype');
     this.data.parentType = isName(parentSubtype) ? parentSubtype.name : null;
-    this.data.parentId = dict.getRaw('Parent').toString();
+    const rawParent = dict.getRaw('Parent');
+    this.data.parentId = rawParent ? rawParent.toString() : null;
 
     if (parentItem.has('RT')) {
-      if ((parentItem.get('RT').name ||
-        AnnotationReplyType.REPLY) === AnnotationReplyType.GROUP) {
-        // If parent item is group reply type,
-        // we have to take data of it's master(IRT)
+      const rt = parentItem.get('RT');
+      if ((isName(rt) ? rt.name : null) === AnnotationReplyType.GROUP) {
```

Comment:
This can be made a lot simpler, if you look at the [definition of the `isName` helper function](https://github.com/mozilla/pdf.js/blob/d0892e46e2b0d16ecd7f720f36f36202ae1c12cd/src/core/primitives.js#L284-L286):
```javascript
if (isName(parentItem.get('RT'), AnnotationReplyType.GROUP)) {
```